realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -59,6 +59,7 @@
             "This Realm instance has already been closed, making it unusable.";
     private static final String NOT_IN_TRANSACTION_MESSAGE =
             "Changing Realm data can only be done from inside a transaction.";
+    static final String LISTENER_NOT_ALLOWED_MESSAGE = "Listeners cannot be used on current thread.";
```

Comment:
I am trying to throw a more general message here since it will be used not only Android. But yes, the information about looper is in the message as well. See `AndroidCapabilities.checkCanDeliverNotification()`.